### PR TITLE
MWPW-143143: Improvements for Upgrade modal

### PR DESCRIPTION
**Pull request by 3ch023**
### Links
Resolves https://jira.corp.adobe.com/browse/MWPW-143143 
Tacocat PR: https://git.corp.adobe.com/wcms/tacocat.js/pull/592

### Summary
* pass promo code, set on Upgrade Offer CTA to the Upgrade Modal
* don't show the Upgrade modal, unless 'upgrade=true' flag is present on Merch Link
* Upgrade modal to not depend on 'entitlement' flag (to enable the case when GWP needs a merch link to have Upgrade feature but not show the TWP/D2P modals).

### How To Test

- login on https://www.stage.adobe.com/ with an eligible test user
- open [test page ](https://main--cc--adobecom.hlx.page/cc-shared/fragments/drafts/commerce/testpage?milolibs=upromo)
- verify CTAs with 'upgrade=true' flag show modal
- re-login with not eligible test user and verify all links lead to Checkout Page

image

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.live/drafts/mariia/pr/price-cta?georouting=off&martech=off
- After: https://upromo--milo--adobecom.hlx.live/drafts/mariia/pr/price-cta?georouting=off&martech=off

### CODE REVIEW

File: libs/blocks/merch/merch.js [hunk 1]
```diff
@@ -226,7 +226,7 @@ export async function getUpgradeAction(
   imsSignedInPromise,
   [{ productArrangement: { productFamily: offerFamily } = {} }],
 ) {
-  if (options.entitlement === false) return undefined;
+  if (!options.upgrade) return undefined;
```
Comment (yesil) [inline libs/blocks/merch/merch.js:229]: don't we need both flags here?

Comment (3ch023) [reply libs/blocks/merch/merch.js:229]: i was thinking about it, but then decided to remove 'entitlement' to have more flexibility.

- entitlement=true&upgrade=true: users will have both TWP modal and upgrade modal experience
- entitlement=true&upgrade=false: users will have only TWP modal 
- entitlement=false&upgrade=true: users will have only Upgrade modal, but no TWP
- entitlement=false&upgrade=false: only checkout page

**Review by mokimo — APPROVED** [review]

File: libs/blocks/merch/merch.js (hunk 1 above)
Comment (yesil) [reply libs/blocks/merch/merch.js:229]: >entitlement=true&upgrade=true: users will have both TWP modal and upgrade modal experience
actually this is not correct, you need the **modal=true**

**Review by yesil — APPROVED** [review]

**Comment by Roycethan** [comment]
@3ch023  Verified

**Comment by 3ch023** [comment]
Adding 'do not merge', since this PR also is blocked by Optical Price audit

**Pull request merged**